Cargo.toml

```diff
@@ -92,6 +93,9 @@ tracing-subscriber = { version = "0.3", features = ["json", "env-filter"] }
 url = { version = "2.5", features = ["serde"] }
 uuid = "1.7"
 which = "6.0"
+openssl = { version = "0.10.68", features = ["vendored"], optional = true }
+openssl-probe = { version = "0.1.5", optional = true }
+git2 = { version = "0.19.0" , features = ["vendored-openssl"] }
```

Comment:
What's the reason for using the `git2` crate instead of the user's installed git binary?

Reply:
I don't want to rely on the user's environment to be honest, also on our side it is easier to manipulate the library compared to system commands

Reply:
Relying on the user's environment in this case might be beneficial because what's the point of initializing a git repo if they do not have git installed?

*easier* to manipulate would also be an overstretch in my humble opinion.

I just wanted to advocate for this solution because it might simplify our CI too. Everything looks good apart from that 😉

Reply:
I also thought about relying on the user's git, but I looked at how cargo handles it, and they do use git2 so I think this solution is fine. In any case it's not a huge deal for someone to have an unnecessary .git directory